cmd/snap-update-ns/system.go

```diff
@@ -47,6 +47,7 @@ func applySystemFstab(instanceName string, fromSnapConfine bool) error {
 		// namespace is locked. This is used by snap-confine to use
 		// snap-update-ns to apply mount profiles.
 		if err := lock.TryLock(); err != osutil.ErrAlreadyLocked {
+			lock.Close()
```

Comment:
Isn't the defer above handling both places already?

Reply:
Indeed! This only became an issue after I switched the logic to return a restore-like function. In this code, before the refactoring, locking is done as a part of larger body of work so there is indeed a defer there.

Reply:
Since the refactored version already has this thing fixed. I can close this.